### PR TITLE
NIFI-6998 PutAzureEventHub to utilize Batch, Properties & Partitioning key

### PR DESCRIPTION
**Pull request by cima**
#### Description of PR

Batch & Partitioning key
+ Asynchronous sending in limited batches
+ Property to determine attribute name carrying partitioning key
+ Maximum batch size property
+ Carrier object

Attributes to User Defined properties

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [x] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on both JDK 8 and JDK 11?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [x] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [x] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [x] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by cima** [comment]
I have corrected the checkstyle violations. I have had difficulties to run complete build of NiFi. It is crashing on _nifi-properties-loader_. So I have been just building the azure processors submodules.

File: nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/test/java/org/apache/nifi/processors/azure/eventhub/PutAzureEventHubTest.java [hunk 1]
```diff
@@ -16,20 +16,58 @@
...
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ScheduledExecutorService;
+
+import org.apache.commons.lang3.time.StopWatch;
```
Comment (sburges) [inline nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/test/java/org/apache/nifi/processors/azure/eventhub/PutAzureEventHubTest.java]: Should this be org.apache.nifi.util.StopWatch

File: nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/test/java/org/apache/nifi/processors/azure/eventhub/PutAzureEventHubTest.java [hunk 2]
```diff
@@ -16,20 +16,58 @@
...
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
 
-import java.util.concurrent.ScheduledExecutorService;
+import com.google.common.collect.ImmutableMap;
+import com.microsoft.azure.eventhubs.EventData;
+import com.microsoft.azure.eventhubs.EventHubClient;
+
+import junit.textui.TestRunner;
```
Comment (sburges) [inline nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/test/java/org/apache/nifi/processors/azure/eventhub/PutAzureEventHubTest.java]: Should this be: org.apache.nifi.util.TestRunner;

**Comment by sburges** [comment]
Outside of the failing PutAzureEventHubTest, the code looks correct. I also verified that functionally this work and messages are being sent to the Partition being specified in the custom attribute.

**Comment by cima** [comment]
You are right. The test was working until I applied organize imports. Unfortunately I assumed that reorganization of imports can't break anything so I have just commited aftewards. I fix it immediately.

**Comment by joewitt** [comment]
@cima When do you think you might have that?  I'm keeping an eye on this from a release manager point of view for NiFi 1.11.  No rush just saying if it is resolved I'd be happy to grab it.

**Comment by cima** [comment]
All right, after build without tests downloaded the entire internet and run out of diskspace (40 GB approx.)I was able to at least run the azure processors project to verify it is compiling and passing test. However I am not going to test the whole NiFi project because there are tests [NiFiPropertiesLoaderGroovyTest.groovy](https://github.com/apache/nifi/blob/master/nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-properties-loader/src/test/groovy/org/apache/nifi/properties/NiFiPropertiesLoaderGroovyTest.groovy#L333) that crashes on windows just because they assume some capability of filesystem. But that is different story. I have pusshed hopefully the last change that should make CI/CD to pass.

**Pull request closed without merging**